passthrough depth occlusion

rosefromthedead opened the pull request:
yay

Foxinatel commented:
Before this gets merged I would quite like to figure out how to combine this with the PR at #26
It's nothing major but both of these branches introduce an identical `oxr!` macro, which isn't ideal
I think the best way to go about this here is to move `depth.rs` into `xr/depth.rs` to mirror `xr/scene.rs` with both of them making use of the macro available at `crate::xr`

rosefromthedead commented:
:warning: I haven't tested the new rev! last time I rebased it broke the game, so don't merge until someone's tested. I didn't change anything scary though. fixed conflicts, moved new files into src/xr/, and fixed a couple of warnings

rosefromthedead commented:
Tested in 4.01 just now!
